sammo.py

```diff
@@ -3,18 +3,48 @@
 __contact__ = "info@hytech-imaging.fr"
 __copyright__ = "Copyright (c) 2021 Hytech Imaging"
 
-from .src.gui.session import SammoActionSession
+from .src.gui.session import SammoActionSession, ParentOfSammoActionSession
+from .src.gui.onOffEffort import SammoActionOnOffEffort, ParentOfSammoActionOnOffEffort
+from .src.core.session import SammoSession
 
 
-class Sammo:
+class Sammo(ParentOfSammoActionSession, ParentOfSammoActionOnOffEffort):
     def __init__(self, iface):
         self.iface = iface
-        self.toolBar = self.iface.addToolBar("Sammo ToolBar")
-        self.actionSession = SammoActionSession(self.iface.mainWindow(), self.toolBar)
+        self._toolBar = self.iface.addToolBar("Sammo ToolBar")
+        self._session = SammoSession()
+        self.actionSession = SammoActionSession(self)
+        self.actionOnOffSession = SammoActionOnOffEffort(self)
 
     def initGui(self):
         self.actionSession.initGui()
+        self.actionOnOffSession.initGui()
 
     def unload(self):
         self.actionSession.unload()
-        del self.toolBar
+        self.actionOnOffSession.unload()
+        del self._toolBar
+
+    def onCreateSession(self, workingDirectory : str):
+        self._session.onCreateSession(workingDirectory)
+        self.actionOnOffSession.onCreateSession()
+
+    def OnStartEffort(self):
+        [feat, table] = self._session.getReadyToAddNewFeatureToEnvironmentTable()
+        if self.iface.openFeatureForm(table, feat):
+            self._session.addNewFeatureToEnvironmentTable(feat)
+
+    def OnStopEffort(self):
+        self._session.onStopEffort()
+
+    @property
+    def mainWindow(self):
+        return self.iface.mainWindow()
+
+    @property
+    def toolBar(self):
+        return self._toolBar
```

Comment:
In this case, `_toolbar` shoudn't be protected.

Reply:
I think it should because there is no need to have a property getter if toolBar is public.
Furthermore, if "self._toolBar" becomes "self.toolBar" there will be a confusion between self.toolBar as property and self.toolBar as a member variable. So if you really want _toolBar to be public, I will choose a another name which will not be very clear.
How would you manage that ?

Reply:
> I think it should because there is no need to have a property getter if toolBar is public.

Yes, that's what I had in mind. `toolBar` should be public SO the property is not necessary.

Reply:
(I know it's not necessarily intuitive to have public member when you come from the c++ world :))